File: Exercicio8/Exercicio8.cpp
#include <iostream>
#include <chrono>

long calcularDiferencaDias(std::tm dataInicial, std::tm dataFinal) {
    std::time_t t1 = std::mktime(&dataInicial);
    std::time_t t2 = std::mktime(&dataFinal);

    if (t1 > t2) {
        throw std::runtime_error("A data inicial não pode ser posterior à data final.");
    }

    return std::difftime(t2, t1) / (60 * 60 * 24);
}

File: Exercicio8/Exercicio8.java
import java.time.LocalDate;
import java.time.temporal.ChronoUnit;

public class Exercicio8 {

    public static long calcularDiferencaDias(LocalDate dataInicial, LocalDate dataFinal) throws Exception {
        if (dataInicial.isAfter(dataFinal)) {
            throw new Exception("A data inicial não pode ser posterior à data final.");
        }
        return ChronoUnit.DAYS.between(dataInicial, dataFinal);
    }
}